packages/wasm-crypto/src/rs/sr25519.rs

```diff
@@ -134,6 +135,23 @@ pub fn ext_sr_verify(signature: &[u8], message: &[u8], pubkey: &[u8]) -> bool {
 	}
 }
 
+/// Key agreement between other's public key and self secret key.
+///
+/// * pubkey: UIntArray with 32 element
+/// * secret: UIntArray with 64 element
+///
+/// * returned vector is the generated secret of 32 bytes.
+#[wasm_bindgen]
+pub fn ext_sr_agree(pubkey: &[u8], secret: &[u8]) -> Vec<u8> {
+		let secret = SecretKey::from_ed25519_bytes(secret).unwrap();
+	    // The first 32 bytes holds the canonical private key
+		let mut key = [0u8; 32];
+		key.copy_from_slice(&secret.to_bytes()[0..32]);
+		let key = Scalar::from_canonical_bytes(key).unwrap();
+		let public = PublicKey::from_bytes(pubkey).unwrap();
+		(&key * public.as_point()).compress().0.to_vec()
```

Comment:
```suggestion
		(&Scalar::from_canonical_bytes(key).unwrap() * PublicKey::from_bytes(pubkey).unwrap().as_point()).compress().0.to_vec()
```

It actually makes an impact on the final size.